Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +352,873 @@ public override bool IsServerConnected()
+            {
+                this.octaneRepository.DeleteEntity<TestSuiteLinkToTests>(GetLoginDTO(), new List<IQueryPhrase>() { qd });
+            });
+        }
+        public string GetLastTestPlanIdFromPath(string path)
+        {
+            string[] separatePath = path.Split('\\');
+
+            separatePath[0] = ExploredApplicationModule.ContainsKey("Application Modules") ? ExploredApplicationModule["Application Modules"] : GetRootFolderId();
+
+            if (!ExploredApplicationModule.ContainsKey("Application Modules"))
+            {
+                ExploredApplicationModule.Add("Application Modules", separatePath[0]);
+            }
+
+            for (int i = 1; i < separatePath.Length; i++)
+            {
+                separatePath[i] = GetTestLabFolderId(separatePath[i], separatePath[i - 1]);
+            }
+
+            return separatePath.Last();
+        }
+
+        public string CreateApplicationModule(string appModuleNameTobeCreated, string desc, string paraentId)
+        {
+            ApplicationModule applicationModule = new ApplicationModule();
+            applicationModule.Name = appModuleNameTobeCreated;
+            applicationModule.SetValue("description", desc);
+
+            applicationModule.SetValue("parent", new BaseEntity("application_module")
+            {
+                Id = paraentId,
+                TypeName = "application_module"
+            });
+
+            ApplicationModule module = Task.Run(() =>
+            {
+                return this.octaneRepository.CreateEntity<ApplicationModule>(GetLoginDTO(), applicationModule, new List<string>() { "path", "id" });
+            }).Result;
+
+            return module.Id.ToString();
+        }
+
+        public bool ExportActivitiesGroupToALM(ActivitiesGroup activitiesGroup, QCTestCase mappedTest, string fatherId, ObservableList<ExternalItemFieldBase> testCaseFields, ObservableList<ExternalItemFieldBase> designStepsFields, ObservableList<ExternalItemFieldBase> designStepsParamsFields, ref string result)
+        {
+            try
+            {
+                string testId;
+                string step = string.Empty;
+                activitiesGroup.ActivitiesIdentifiers.ToList().ForEach(p =>
+                {
+                    if (!string.IsNullOrEmpty(step))
+                    {
+                        step += "- ";
+                    }
+                    step += p.ActivityName + " ";
+
+                    if (p.IdentifiedActivity.Variables.Any())
+                    {
+                        p.IdentifiedActivity.Variables.ToList().ForEach(f =>
+                        {
+                            step += " <" + f.Name + "> ";
+                        });
+                    }
+                    step += '\n';
+                });
+
+                if (mappedTest == null) //#Create new test case
+                {
+
+                    testId = CreateNewTestCase(activitiesGroup, fatherId, testCaseFields, step);
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'testId'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)